### DISCUSSION TITLE
gtk4-layer-shell

### DISCUSSION THREAD

**Discussion by aquifolly** [JakeStanger/ironbar#101] [General]
Is there any plans to switch to https://github.com/wmww/gtk4-layer-shell when it stabilises?

**Comment by JakeStanger** [comment]
Eventually yes. I'll wait for https://github.com/wmww/gtk-layer-shell/issues/37 to be completed but I suspect it will take quite a bit of work to get us there, since I'll likely need to update a lot of GTK3 code in Ironbar.

**Reply by yavko** [reply]
I just remembered to comment, but It hit a `1.0` release last week

**Reply by JakeStanger** [reply]
Oh awesome, I'll start having a play soon. It looks like there's a [crate](https://crates.io/crates/gtk4-layer-shell) for it too, so I can start having a look at this soon.

**Comment by JakeStanger** [comment]
I've started work on this, there's a lot of breaking changes and the libraries are in quite an early state that means it's not super straightforward to get up and running, so don't expect anything for a good while. Very early stages (nowhere near building yet) but I'm tracking progress here: https://github.com/JakeStanger/ironbar/pull/112
